packages/core/src/index.ts

```diff
@@ -203,6 +213,37 @@ export class Abby<
     return this.getProjectData();
   }
 
+  /**
+   * Helper function to retrieve the time a flag is valid
+   * @param key 
+   * @returns 
+   */
+  getFeatureFlagTimeout<F extends FlagName>(key: F) {
+    return this.#flagTimeoutMap.get(key)
+  }
+
+  /**
+   * Helper function to check if a featureflag should be refetched
+   * @param key name of the featureflag
+   * @returns value of flag
+   */
+  getValidFlag<F extends FlagName>(key: F) {
+    const flagTime = this.#flagTimeoutMap.get(key)
+    if (!flagTime) return this.#data.flags[key];
+    const now = new Date();
+    if (flagTime.getTime() <= now.getTime()) {
+      this.refetchFlags()
```

Comment:
this is async.. so without waiting the return down below may be old though :shrug: